### ISSUE TITLE
Allow unauthenticated usage for public API endpoints

### ISSUE THREAD

**Issue by StephenKing** [terraform-providers/terraform-provider-github#205]
Hi there,

### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

### Affected Resource(s)
Please list the resources as a list, for example:
- data.github_ip_ranges
- data.github_repositories
- data.github_* actually..

### Terraform Configuration Files
```hcl
provider "github" {
  organization = "" # required setting
  token = "" # required setting
}

data "github_ip_ranges" "this" {}

output "github_ip_ranges_hooks" {
  value = "${data.github_ip_ranges.this.hooks}"
}
```

### Expected Behavior

The API endpoint https://api.github.com/meta can be reached without authentication. It would be easier to use this data source (or the github data sources in general), if the provider wouldn't enforce that a token is given.

I know that rate limiting can kick in a lot earlier then. But for the use case of white-listing the IP addresses that GitHub uses to deliver hooks, our Terraform deployment would be easier and the number of calls sufficiently low.

### Actual Behavior

I'm asked to provide a Github token.

**Comment by majormoses** [comment]
I think this is a duplicate of https://github.com/terraform-providers/terraform-provider-github/issues/190

**Comment by megan07** [comment]
Thanks @StephenKing for opening this issue. It's been closed with #255 and will be released shortly.
